src/Shared/ServerInfrastructure/StringUtilities.cs

```diff
@@ -17,102 +18,211 @@ internal class StringUtilities
+                            return false;
                         }
-                        if (input <= end - sizeof(int))
-                        {
-                            isValid &= CheckBytesInAsciiRange(((int*)input)[0]);
 
-                            output[0] = (char)input[0];
-                            output[1] = (char)input[1];
-                            output[2] = (char)input[2];
-                            output[3] = (char)input[3];
+                        var tmp0 = Avx2.UnpackLow(vector, zero);
+                        var tmp1 = Avx2.UnpackHigh(vector, zero);
 
-                            input += sizeof(int);
-                            output += sizeof(int);
-                        }
+                        // Bring into the right order
+                        var out0 = Avx2.Permute2x128(tmp0, tmp1, 0x20);
+                        var out1 = Avx2.Permute2x128(tmp0, tmp1, 0x31);
+
+                        Avx.Store((ushort*)output, out0.AsUInt16());
+                        Avx.Store((ushort*)output + Vector256<ushort>.Count, out1.AsUInt16());
+
+                        input += Vector256<sbyte>.Count;
+                        output += Vector256<sbyte>.Count;
+                    } while (input <= end - Vector256<sbyte>.Count);
+
+                    if (input == end)
+                    {
+                        return true;
                     }
-                    else
+                }
+
+                if (input <= end - Vector128<sbyte>.Count)
+                {
+                    Vector128<sbyte> zero = Vector128<sbyte>.Zero;
+
+                    do
                     {
-                        // 32-bit: Loop ints by default
-                        while (input <= end - sizeof(int))
+                        var vector = Sse2.LoadVector128(input).AsSByte();
+                        if (!CheckBytesInAsciiRange(vector, zero))
                         {
-                            isValid &= CheckBytesInAsciiRange(((int*)input)[0]);
+                            return false;
+                        }
 
-                            output[0] = (char)input[0];
-                            output[1] = (char)input[1];
-                            output[2] = (char)input[2];
-                            output[3] = (char)input[3];
+                        var c0 = Sse2.UnpackLow(vector, zero).AsUInt16();
+                        var c1 = Sse2.UnpackHigh(vector, zero).AsUInt16();
 
-                            input += sizeof(int);
-                            output += sizeof(int);
-                        }
+                        Sse2.Store((ushort*)output, c0);
+                        Sse2.Store((ushort*)output + Vector128<ushort>.Count, c1);
+
+                        input += Vector128<sbyte>.Count;
+                        output += Vector128<sbyte>.Count;
+                    } while (input <= end - Vector128<sbyte>.Count);
+
+                    if (input == end)
```

Comment:
Does only helps if the decodes string happens to have a multiple of 16 characters, right? Seems unlikely to be worth it on average.

Reply:
Is a single check avoids the 4 check chain: `<= sizeof(long)`, `<= sizeof(int)`, `<= sizeof(char)`, `<= 1`; though suppose it adds one more to everything else.

Reply:
Yes that's right. I tend to keep it, as @benaadams explained, and I think it doesn't harm on input-length that are not multiples of 16 chars long.